packages/web/src/shared/AwesomeFade.tsx

```diff
@@ -0,0 +1,72 @@
+import { keyframes } from '@emotion/core'
+import * as React from 'react'
+import Reveal from 'react-awesome-reveal'
+
+type Direction = 'up' | 'right'
+
+interface Props {
+  children: React.ReactNode
+  delay?: number
+  distance?: number | string
+  duration?: number
+  fraction?: number
+  reverse?: boolean
+  direction?: Direction
+}
+
+export default React.memo(function AwesomeFade({
+  children,
+  reverse,
+  delay,
+  distance,
+  direction,
+  duration,
+  fraction,
+}: Props) {
+  return (
+    <Reveal
+      keyframes={React.useMemo(() => getKeyFrames(distance, direction, reverse), [
+        direction,
+        distance,
+      ])}
+      triggerOnce={false}
+      delay={delay}
+      duration={duration}
+      fraction={fraction}
+    >
+      {children}
+    </Reveal>
+  )
+})
+
+function getKeyFrames(distance: number | string, direction: Direction, reverse: boolean) {
+  let from = `translate3d(0, ${distance}, 0)`
+
+  if (direction === 'right') {
+    from = `translate3d(${distance}, 0, 0)`
+  }
+
+  if (reverse) {
+    return keyframes`
+    from {
+      opacity: 1;
+      transform: translate3d(0, 0, 0);
+    }
+    to {
+      opacity: 0;
+      transform: ${from}; 
+    }
+  `
+  }
+
+  return keyframes`
+    from {
+      opacity: 0;
+      transform: ${from};
+    }
+    to {
+      opacity: 1;
+      transform: translate3d(0, 0, 0);
+    }
+  `
+}
```

Comment:
Would like to shorten this to a single return statement using variables, but I don't have any good ideas...
``` 
if (reverse) {
    return keyframes`
    from {
      opacity: 1;
      transform: translate3d(0, 0, 0);
    }
    to {
      opacity: 0;
      transform: ${from}; 
    }
  `
  }

  return keyframes`
    from {
      opacity: 0;
      transform: ${from};
    }
    to {
      opacity: 1;
      transform: translate3d(0, 0, 0);
    }
  `
```

Reply:
```
const to = 'translate3d(0, 0, 0)`

return keyframes`
  from {
    opacity: ${+reverse};
    transform: ${reverse ? to : from};
  } to {
    opacity: ${+(!reverse)};
    transform: ${reverse ? from : to};
  }
`
```
Eh, that's kind of what I was thinking, but makes it harder to read.  It's probably fine as is :)

Reply:
oh interesting using the coersion power of `+` to turn boolean in to 1 or 0. clever

Reply:
i like it